draft-ietf-oauth-v2-1.md

```diff
@@ -431,7 +431,7 @@ The flow illustrated in {{fig-refresh-token-flow}} includes the following steps:
 ### Client Credentials
 
 The client credentials or other forms of client authentication
-(e.g. a `client_secret` or a private key used to sign a JWT)
+(e.g. a private key used to sign a JWT)
```

Comment:
@tlodderstedt, I'm wondering if there are any concerns with `client_secret_jwt` as a client auth type.

Reply:
It requires both client and AS to know the secret.

Reply:
At some degree `client_secret_basic` are `client_secret_post` are better then `client_secret_jwt` in this regard as the AS may store only client_secrets' hashes. This is the idea, right?

Maybe it makes sense to mention/specify `private_key_jwt` as the client authentification form behind

> a private key used to sign a JWT.

Reply:
Probably is worth mentioning RFC7523 for this example.

Reply:
I suggest also changing a reference from OpenID to RFC7523 in the following sentences

> Some additional authentication methods such as mTLS {{RFC8705}} and "private_key_jwt" {{OpenID}}


and

`It is RECOMMENDED to use asymmetric (public-key based) methods for client authentication such as mTLS {{RFC8705}} or "private_key_jwt" {{OpenID}}.`

Reply:
Agreed, I'll add these changes in a separate commit.